e2e/pages/maps/planting.py

```diff
@@ -44,67 +73,171 @@ def close_tour(self):
+        )
+        self._page.wait_for_timeout(300)
 
     def click_delete(self):
         """Deletes a planting by clicks on the delete button."""
-        self.delete_plant_button.click()
+        self._delete_plant_button.click()
 
     def click_undo(self):
         """Clicks the undo button."""
-        self.undo_button.click()
+        self._undo_button.click()
 
     def click_redo(self):
         """Clicks the redo button."""
-        self.redo_button.click()
+        self._redo_button.click()
+
+    def select_birdie_background(self):
+        """
+        Selects the Birdie.jpg background image
+        since its the only image allowed to load (see `dont_load_images()`).
+        Performs a short delay.
+        """
+        self._background_button.click()
+        self._page.get_by_text("Birdie.jpg").click()
+        # Delay so image can be rendered on canvas
+        self._page.wait_for_timeout(2000)
+
+    def click_hide_base_layer(self):
+        self._hide_base_layer.click()
+
+    def click_hide_plant_layer(self):
+        self._hide_plant_layer.click()
+        self._page.wait_for_timeout(5000)
 
-    def expect_plant_to_be_planted(self, plant_name):
+    def screenshot_canvas(self, timeout=300):
+        """
+        Takes a grayscale screenshot of the canvas.
+        Has a default 300ms delay before taking the screenshot,
+        to ensure everything is stable.
+
+        Parameters
+        ----------
+        timeout : int, optional, default=500
+        Timeout in ms before taking the screenshot.
+
+        Returns
+        -------
+        opencv2 image object of the screenshot
+        """
+        self._page.wait_for_timeout(timeout)
+        buffer = self._canvas.screenshot()
+        self._screenshot = cv2.imdecode(
+            np.frombuffer(buffer, dtype=np.uint8), cv2.IMREAD_GRAYSCALE
+        )
+        return self._screenshot
+
+    def expect_canvas_equals_last_screenshot(self, test_name, rtol=0, atol=5):
+        """
+        This method compares the last taken screenshot of the canvas
+        with a new one.
+        Use `mpp.canvas_screenshot()` before to take a screenshot.
+        If the comparison fails two .png images are saved.
+
+        Parameters
+        ----------
+        test_name : str
+        The name of the image files if the test fails.
+        Best to pass the test name from pytests fixture `request`.
+        atol : float, optional, default=5
+        Absolut tolerance.
+        rtol : float, optional, default=0
+        Relative tolerance.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        `AssertionError` when the two screenshots are not equal.
+
+        Examples
+        --------
+        mpp.assert_canvas_equals_last_screenshot(request.node.name)
+        """
+        expected = self._screenshot
+        actual = self.screenshot_canvas()
+        try:
+            np.testing.assert_allclose(expected, actual, rtol=rtol, atol=atol)
```

Comment:
Good point, this will just make sure the tests are not too flaky on that option in case some rounding errors happen or the difference is very small. So from API perspective it could be renamed to close, but its still very very very close, there is not much room for different images. So close could be misleading too, as the relative difference is 0 and the absolute is 5px.